File: src/main/java/no/nav/samordning/hendelser/feed/FeedExceptionHandler.java
package no.nav.samordning.hendelser.feed;

import org.springframework.http.HttpStatus;
import org.springframework.http.ResponseEntity;
import org.springframework.web.bind.annotation.ControllerAdvice;
import org.springframework.web.bind.annotation.ExceptionHandler;

@ControllerAdvice
public class FeedExceptionHandler {

    @ExceptionHandler(BadParameterException.class)
    public ResponseEntity<String> handleException(BadParameterException e) {
        return ResponseEntity
                .status(HttpStatus.BAD_REQUEST)
                .body(e.getMessage());
    }
}

File: src/main/kotlin/no/nav/samordning/hendelser/feed/FeedExceptionHandler.kt
package no.nav.samordning.hendelser.feed

import org.springframework.http.HttpStatus.BAD_REQUEST
import org.springframework.web.bind.annotation.ControllerAdvice
import org.springframework.web.bind.annotation.ExceptionHandler
import org.springframework.web.bind.annotation.ResponseStatus
import javax.validation.ConstraintViolationException

@ControllerAdvice
class FeedExceptionHandler {

    @ExceptionHandler(ConstraintViolationException::class)
    @ResponseStatus(BAD_REQUEST)
    fun handleException(e: ConstraintViolationException) = e.message
}
